src/Components/PostDisplay.jsx

```diff
@@ -0,0 +1,144 @@
+import { useState, useEffect } from "react";
+import { onChildAdded, push, ref, set, remove, onChildRemoved } from "firebase/database";
+import { signOut } from "firebase/auth";
+import { database, storage } from "./firebase";
+import { ref as storageRef, uploadBytes, getDownloadURL } from "firebase/storage";
+import { auth } from "./firebase.jsx";
+
+export default function PostDisplay({ user, setUser, setIsLoggedIn, IsLoggedIn }) {
+  const [inputTextValue, setInputTextValue] = useState("");
+  const [fileInputFile, setFileInputFile] = useState(null);
+  const [messages, setMessages] = useState([]);
+
+  const DB_MESSAGES_KEY = "messages";
+  const DB_STORAGE_KEY = "images";
+  const currentDate = new Date().toLocaleString();
+
+  useEffect(() => {
+    const messagesRef = ref(database, DB_MESSAGES_KEY);
+    onChildAdded(messagesRef, (data) => {
+      setMessages((prevState) => [...prevState, { key: data.key, val: data.val() }]);
+    });
+    onChildRemoved(messagesRef, (data) => {
+      setMessages((prevState) => prevState.filter((item) => item.key !== data.key));
+    });
+  }, []);
+
+  const writeData = async (e) => {
+    e.preventDefault();
+    const messageListRef = ref(database, DB_MESSAGES_KEY);
+    const newMessageRef = push(messageListRef);
+    const storageRefInstance = storageRef(storage, DB_STORAGE_KEY + (fileInputFile ? fileInputFile.name : ""));
+    try {
+      if (fileInputFile) {
+        console.log("File Input:", fileInputFile);
+        await uploadBytes(storageRefInstance, fileInputFile);
+        const url = await getDownloadURL(storageRefInstance);
+        set(newMessageRef, {
+          Comment: inputTextValue,
+          Timestamp: currentDate,
+          url: url,
+        });
+      } else {
+        set(newMessageRef, {
+          Comment: inputTextValue,
+          Timestamp: currentDate,
+        });
+      }
+    } catch (err) {
+      console.log("Error:", err);
+    }
+    setInputTextValue("");
+    setFileInputFile(null);
+  };
+
+  const deleteMessage = (messageKey) => {
+    const updatedMessageList = messages.filter((message) => message.key !== messageKey);
+    setMessages(updatedMessageList);
+    remove(ref(database, `${DB_MESSAGES_KEY}/${messageKey}`));
+  };
+
+  const handleSignOut = () => {
+    signOut(auth).then(() => {
+      setIsLoggedIn(false);
+      setUser({});
+    });
+  };
+  console.log("setIsLoggedIn:", setIsLoggedIn);
+
+  return (
+    <div className="card">
+      {Object.keys(user).length !== 0 ? <p> welcome, {user.email} </p> : null}
+      <br />
+      {IsLoggedIn ? (
```

Comment:
Technically would work better on the navigation bar, either show logout button or signup/ signin button.
Nice utilisation of tailwind CSS properties